Python/pymath.c

```diff
@@ -79,3 +79,18 @@ round(double x)
     return copysign(y, x);
 }
 #endif /* HAVE_ROUND */
+
+static const unsigned int BitLengthTable[32] = {
+    0, 1, 2, 2, 3, 3, 3, 3, 4, 4, 4, 4, 4, 4, 4, 4,
+    5, 5, 5, 5, 5, 5, 5, 5, 5, 5, 5, 5, 5, 5, 5, 5
+};
+
+unsigned int _Py_bit_length(unsigned long d) {
```

Comment:
Would it make sense to convert this function to an static inline function only exposed in the internal C API?

Reply:
For sure, the declaration in `pymath.h` should be protected by an `#ifndef Py_LIMITED_API`. @vstinner: I'm not sure I'm understanding you correctly, though. Making this `static` would make it unavailable to the other files that need it, wouldn't it?

Reply:
Ah, I see: you're suggesting moving the entire function definition to `pymath.h`, and making it `static inline`. Is that right?

Reply:
Right. Sorry, my comment was unclear. I converted some macros to static inline functions in header files.

I don't know if it would be worth it in term of performance.

Reply:
Could be worth a try; I'd be surprised if it makes a measurable difference, but we won't know without testing.

But if not, we do need the `Py_LIMITED_API` guard: we don't want to expose `_Py_bit_length` to 3rd party libraries.